src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripDesigner.cs

```diff
@@ -1145,8 +1154,25 @@ protected override void Dispose(bool disposing)
                 _items = null;
             }
 
+            if (_undoEngine is not null)
+            {
+                _undoEngine.Undoing -= new EventHandler(OnUndoing);
```

Comment:
```suggestion
                _undoEngine.Undoing -= OnUndoing;
```
and the same on the next line.